kolibri/plugins/epub_viewer/assets/src/views/EpubRendererIndex.vue

```diff
@@ -678,6 +703,7 @@
         this.sliderValue = location.start.percentage * 100;
         this.updateCurrentSection(location.start);
         this.currentLocation = location.start.cfi;
+        this.listVisitedPages(this.currentLocation);
```

Comment:
Hrm, the `listVisitedPages` method updates the returned value of the computed `this.visitedPages` property - this is different to setting `this.currentLocation` which is data, and so is local state that is being updated.

This means that when you then call `this.updateContentState()` the updates you have made to `this.visitedPages` will only be present because Vue has cached the output of the `visitedPages` computed property that you have then modified. If the computed property were to be recomputed for some reason, then your modifications would be lost.

Two possible solutions:

Do what currentLocation and savedLocation do - have one in data, and one that is a computed prop derived from extraFields.contentState.

Pass the `currentLocation` to `updatedContentState` and just update `visitedPages` on the `contentState`, and then rely on that being set into the `extraFields` prop from outside the component, to then update the computed prop `visitedPages`